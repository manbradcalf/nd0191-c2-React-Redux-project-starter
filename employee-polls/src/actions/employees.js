export const RECEIVE_EMPLOYEES = "RECEIVE_EMPLOYEES";

export function receiveEmployees(employees) {
  return {
    type: RECEIVE_EMPLOYEES,
    employees
  }
}
