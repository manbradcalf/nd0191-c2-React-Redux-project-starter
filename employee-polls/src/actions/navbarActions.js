export const SET_SELECTED_PAGE = 'SET_SELECTED_PAGE';

export function setSelectedPage(selectedPage) {
  return {
    type: SET_SELECTED_PAGE,
    selectedPage,
  };
}
