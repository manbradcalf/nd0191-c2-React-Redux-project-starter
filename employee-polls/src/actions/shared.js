import { getInitialData } from "../util/api";
import { receiveEmployees } from "./employees";
import { receiveQuestions } from "./questions";
import { setAuthedUser } from "./authedUser";

// todo: replace with auth functionality
const AUTHED_ID = "sarahedo";

export function handleInitialData() {
  return (dispatch) => {
    return getInitialData()
      .then(({ employees, questions }) => {
        dispatch(receiveQuestions(questions));
        dispatch(receiveEmployees(employees));
        dispatch(setAuthedUser(AUTHED_ID));
      })
      .catch((e) => {
        console.log(e);
      });
  };
}
