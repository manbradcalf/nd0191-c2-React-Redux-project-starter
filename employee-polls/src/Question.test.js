import { render, fireEvent, getByTestId } from "@testing-library/react";
import React from "react";
import Question from "./components/Question";
import configureStore from "redux-mock-store";
import { exampleState } from "./exampleState";
import middleware from "./middleware";
import reducer from "./reducers";
import { Provider } from "react";
import { handleQuestionAnswered } from "./actions/shared";

const mockStore = configureStore([]);

describe("Question", () => {
  it("will match the snapshot when example state is passed", () => {
    var component = render(<Question store={mockStore(exampleState)} />);
    expect(component).toMatchSnapshot();
  });

  it("matches the snapshot when no data is passed", () => {
    var component = render(<Question store={mockStore({})} />);
    expect(component).toMatchSnapshot();
  });

  it("matches the snapshot when part of the data is missing", () => {
    let mockState = exampleState;
    mockState.employees = {};
    let store = mockStore(mockState);
    var component = render(<Question store={mockStore(mockState)} />);

    expect(component).toMatchSnapshot();
  });

  it("Dispatches event when answer selected", () => {
    let mockState = exampleState;
    let store = mockStore(mockState);

    store.dispatch = jest.fn();

    var component = render(<Question store={store} />);
    var voteOptionOneBtn = component.getByTestId("voteOptionOne");
    fireEvent.click(voteOptionOneBtn);
    expect(store.dispatch).toHaveBeenCalledTimes(1);
  });
});
