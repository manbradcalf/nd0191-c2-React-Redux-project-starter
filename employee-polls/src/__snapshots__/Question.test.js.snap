// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionDetailPage matches the snapshot when no data is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-19midj6"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1rs1pcl-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="https://www.svgrepo.com/show/163025/avatar.svg"
              />
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1gy3ww3-MuiTypography-root"
            >
              tylermcginnis
               asked...
            </p>
          </div>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
          >
            Would you rather?
          </h4>
          <div
            class="MuiBox-root css-308nzh"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
            >
              take a course on ReactJS
            </p>
            <span
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="50"
              class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
              role="progressbar"
            >
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
                style="transform: translateX(-50%);"
              />
            </span>
            <div
              class="MuiBox-root css-0"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              >
                Votes [
                1
                ]:
                 
                tylermcginnis
              </span>
              <br />
              <span
                class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              >
                50
                %
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root css-1sya07t"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              take a course on unit testing with Jest
            </p>
            <span
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="50"
              class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
              role="progressbar"
            >
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
                style="transform: translateX(-50%);"
              />
            </span>
            <div
              class="MuiBox-root css-0"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              >
                Votes [
                1
                ]:
                 
                mtsamis
              </span>
              <br />
              <span
                class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              >
                50
                %
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root css-1n2mv2k"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
              data-testid="voteOptionOne"
              disabled=""
              tabindex="-1"
              type="button"
              value="optionOne"
            >
              Option 1
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
              data-testid="voteOptionTwo"
              disabled=""
              tabindex="-1"
              type="button"
              value="optionTwo"
            >
              Option 2
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-19midj6"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1rs1pcl-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://www.svgrepo.com/show/163025/avatar.svg"
            />
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gy3ww3-MuiTypography-root"
          >
            tylermcginnis
             asked...
          </p>
        </div>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
        >
          Would you rather?
        </h4>
        <div
          class="MuiBox-root css-308nzh"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
          >
            take a course on ReactJS
          </p>
          <span
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="50"
            class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
            role="progressbar"
          >
            <span
              class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
              style="transform: translateX(-50%);"
            />
          </span>
          <div
            class="MuiBox-root css-0"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              Votes [
              1
              ]:
               
              tylermcginnis
            </span>
            <br />
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              50
              %
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root css-1sya07t"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            take a course on unit testing with Jest
          </p>
          <span
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="50"
            class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
            role="progressbar"
          >
            <span
              class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
              style="transform: translateX(-50%);"
            />
          </span>
          <div
            class="MuiBox-root css-0"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              Votes [
              1
              ]:
               
              mtsamis
            </span>
            <br />
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              50
              %
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root css-1n2mv2k"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
            data-testid="voteOptionOne"
            disabled=""
            tabindex="-1"
            type="button"
            value="optionOne"
          >
            Option 1
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
            data-testid="voteOptionTwo"
            disabled=""
            tabindex="-1"
            type="button"
            value="optionTwo"
          >
            Option 2
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`QuestionDetailPage matches the snapshot when part of the data is missing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-19midj6"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1rs1pcl-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1gy3ww3-MuiTypography-root"
            >
              tylermcginnis
               asked...
            </p>
          </div>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
          >
            Would you rather?
          </h4>
          <div
            class="MuiBox-root css-1sya07t"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
            >
              take a course on ReactJS
            </p>
            <span
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="50"
              class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
              role="progressbar"
            >
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
                style="transform: translateX(-50%);"
              />
            </span>
          </div>
          <div
            class="MuiBox-root css-1sya07t"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              take a course on unit testing with Jest
            </p>
            <span
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="50"
              class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
              role="progressbar"
            >
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
                style="transform: translateX(-50%);"
              />
            </span>
          </div>
          <div
            class="MuiBox-root css-1n2mv2k"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
              data-testid="voteOptionOne"
              tabindex="0"
              type="button"
              value="optionOne"
            >
              Option 1
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
              data-testid="voteOptionTwo"
              tabindex="0"
              type="button"
              value="optionTwo"
            >
              Option 2
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-19midj6"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1rs1pcl-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gy3ww3-MuiTypography-root"
          >
            tylermcginnis
             asked...
          </p>
        </div>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
        >
          Would you rather?
        </h4>
        <div
          class="MuiBox-root css-1sya07t"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
          >
            take a course on ReactJS
          </p>
          <span
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="50"
            class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
            role="progressbar"
          >
            <span
              class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
              style="transform: translateX(-50%);"
            />
          </span>
        </div>
        <div
          class="MuiBox-root css-1sya07t"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            take a course on unit testing with Jest
          </p>
          <span
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="50"
            class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
            role="progressbar"
          >
            <span
              class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
              style="transform: translateX(-50%);"
            />
          </span>
        </div>
        <div
          class="MuiBox-root css-1n2mv2k"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
            data-testid="voteOptionOne"
            tabindex="0"
            type="button"
            value="optionOne"
          >
            Option 1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
            data-testid="voteOptionTwo"
            tabindex="0"
            type="button"
            value="optionTwo"
          >
            Option 2
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`QuestionDetailPage will match the snapshot when example state is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-19midj6"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1rs1pcl-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="https://www.svgrepo.com/show/163025/avatar.svg"
              />
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1gy3ww3-MuiTypography-root"
            >
              tylermcginnis
               asked...
            </p>
          </div>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
          >
            Would you rather?
          </h4>
          <div
            class="MuiBox-root css-308nzh"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
            >
              take a course on ReactJS
            </p>
            <span
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="50"
              class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
              role="progressbar"
            >
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
                style="transform: translateX(-50%);"
              />
            </span>
            <div
              class="MuiBox-root css-0"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              >
                Votes [
                1
                ]:
                 
                tylermcginnis
              </span>
              <br />
              <span
                class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              >
                50
                %
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root css-1sya07t"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              take a course on unit testing with Jest
            </p>
            <span
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="50"
              class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
              role="progressbar"
            >
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
                style="transform: translateX(-50%);"
              />
            </span>
            <div
              class="MuiBox-root css-0"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              >
                Votes [
                1
                ]:
                 
                mtsamis
              </span>
              <br />
              <span
                class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              >
                50
                %
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root css-1n2mv2k"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
              data-testid="voteOptionOne"
              disabled=""
              tabindex="-1"
              type="button"
              value="optionOne"
            >
              Option 1
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
              data-testid="voteOptionTwo"
              disabled=""
              tabindex="-1"
              type="button"
              value="optionTwo"
            >
              Option 2
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-19midj6"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1rs1pcl-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://www.svgrepo.com/show/163025/avatar.svg"
            />
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gy3ww3-MuiTypography-root"
          >
            tylermcginnis
             asked...
          </p>
        </div>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
        >
          Would you rather?
        </h4>
        <div
          class="MuiBox-root css-308nzh"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
          >
            take a course on ReactJS
          </p>
          <span
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="50"
            class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
            role="progressbar"
          >
            <span
              class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
              style="transform: translateX(-50%);"
            />
          </span>
          <div
            class="MuiBox-root css-0"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              Votes [
              1
              ]:
               
              tylermcginnis
            </span>
            <br />
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              50
              %
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root css-1sya07t"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            take a course on unit testing with Jest
          </p>
          <span
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="50"
            class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-1tygqtr-MuiLinearProgress-root"
            role="progressbar"
          >
            <span
              class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
              style="transform: translateX(-50%);"
            />
          </span>
          <div
            class="MuiBox-root css-0"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              Votes [
              1
              ]:
               
              mtsamis
            </span>
            <br />
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              50
              %
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root css-1n2mv2k"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
            data-testid="voteOptionOne"
            disabled=""
            tabindex="-1"
            type="button"
            value="optionOne"
          >
            Option 1
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
            data-testid="voteOptionTwo"
            disabled=""
            tabindex="-1"
            type="button"
            value="optionTwo"
          >
            Option 2
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
